### PR TITLE
Support GUI configuration and improve UI response

### PR DESCRIPTION
**Pull request by mill1000**
This original goal of this PR  was to implement a config flow to allow the integration to be configured via the GUI.

When configured via the GUI a new device with a media player entity is now created.

I got a bit out of hand and I overhauled a number of other things
- Removed usage of deprecated consts in favor of enums
- Fixed an issue with unsubscribing
- Preemptively updated local media player state instead of waiting for the topic. I found that shairport sync does not immediately send an "play_end" but instead a "play_flush" when paused and the lag was annoying.

### CODE REVIEW

File: custom_components/shairport_sync/media_player.py [hunk 1]
```diff
@@ -8,38 +8,23 @@
...
-    TOP_LEVEL_TOPIC_ARTIST,
-    TOP_LEVEL_TOPIC_ALBUM,
-    TOP_LEVEL_TOPIC_COVER,
-    TOP_LEVEL_TOPIC_PLAY_END,
-    TOP_LEVEL_TOPIC_PLAY_START,
-    TOP_LEVEL_TOPIC_REMOTE,
-    TOP_LEVEL_TOPIC_TITLE,
-)
+
+from .const import *
```
Comment (parautenbach) [inline custom_components/shairport_sync/media_player.py:27]: Minor nitpick, but I try to avoid wildcard imports.

**Comment by parautenbach** [comment]
This is awesome, @mill1000! 

On the surface, this all looks good and is a nice catch-up on some Python (annotations) and HA updates. I've been thinking about implementing the config flow stuff, but I'm super short on time. 

If you could also update the README to state which version you've tested with, I'd be happy to merge this. My own testing has just been against the one version I typically run. 

My system using this is currently shut down, but I'll spin it up again soon (and probably upgrade everything on the Pi), which means I'll be in a better position to work on this project again and to test stuff. 

For now, if we can just state which is the known version this was tested again, we're good to go (HA and Shairport Sync's versions).

File: custom_components/shairport_sync/media_player.py (hunk 1 above)
Comment (mill1000) [reply custom_components/shairport_sync/media_player.py]: Switched to using a StrEnum for those types and changed the import to be explicit. Best of both worlds.

**Comment by mill1000** [comment]
@parautenbach, I think this is good to go know.

**Comment by parautenbach** [comment]
Muchas gracias!

**Pull request merged**